File: app/src/main/java/com/thinkzi/weather/ui/view/activity/WeatherInformationUIModelListActivity.java
package com.thinkzi.weather.ui.view.activity;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.Observer;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.thinkzi.weather.ui.R;
import com.thinkzi.weather.ui.databinding.ActivityListItemBinding;
import com.thinkzi.weather.ui.model.WeatherInformationUIModel;
import com.thinkzi.weather.ui.view.adapter.WeatherInformationUIModelListAdapter;
import com.thinkzi.weather.ui.viewmodel.WeatherInCityUIModelForFiveDaysViewModel;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

public class WeatherInformationUIModelListActivity extends BaseActivity {

    @Inject
    WeatherInCityUIModelForFiveDaysViewModel _weatherInCityUIModelForFiveDaysViewModel;

    // android data binding
    private ActivityListItemBinding _activityListItemBinding;

    private WeatherInformationUIModelListAdapter _weatherInformationUIModelListAdapter;

    // intent contain source context and destination context (this Activity) for Navigator
    public static Intent getCallingIntent(Context _context) {
        return new Intent(_context, WeatherInformationUIModelListActivity.class);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // inject dependancy for this Activity
        this.getApplicationComponent().inject(this);

        // android data binding
        _activityListItemBinding = DataBindingUtil.setContentView(this, R.layout.activity_list_item);

        _activityListItemBinding.setLifecycleOwner(this);

        // set viewmodel for databiding
        _activityListItemBinding.setWeatherInCityUIModelForFiveDaysViewModel(_weatherInCityUIModelForFiveDaysViewModel);

        _weatherInformationUIModelListAdapter = new WeatherInformationUIModelListAdapter(getApplicationContext(), new ArrayList<WeatherInformationUIModel>());

        // set adapter for RecyclerView
        _activityListItemBinding.rcvItemList.setLayoutManager(new LinearLayoutManager(WeatherInformationUIModelListActivity.this));
        _activityListItemBinding.rcvItemList.setAdapter(_weatherInformationUIModelListAdapter);

        _weatherInCityUIModelForFiveDaysViewModel.getWeatherInformationUIModelListOfSelectedDateMutableLiveData().observe(WeatherInformationUIModelListActivity.this, new Observer<List<WeatherInformationUIModel>>() {
            @Override
            public void onChanged(List<WeatherInformationUIModel> weatherInformationUIModels) {
                _weatherInformationUIModelListAdapter.setWeatherInformationUIModelList(weatherInformationUIModels);
            }
        });

    }
}

File: app/src/main/java/com/thinkzi/weather/ui/view/activity/WeatherInformationUIModelListActivity.kt
package com.thinkzi.weather.ui.view.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.LinearLayoutManager

import com.thinkzi.weather.ui.R
import com.thinkzi.weather.ui.databinding.ActivityListItemBinding
import com.thinkzi.weather.ui.model.WeatherInformationUIModel
import com.thinkzi.weather.ui.view.adapter.WeatherInformationUIModelListAdapter
import com.thinkzi.weather.ui.viewmodel.WeatherInCityUIModelForFiveDaysViewModel

import java.util.ArrayList

import javax.inject.Inject

class WeatherInformationUIModelListActivity : BaseActivity() {

    @Inject
    lateinit var _weatherInCityUIModelForFiveDaysViewModel: WeatherInCityUIModelForFiveDaysViewModel

    // android data binding
    private var _activityListItemBinding: ActivityListItemBinding? = null

    private var _weatherInformationUIModelListAdapter: WeatherInformationUIModelListAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // inject dependancy for this Activity
        this.applicationComponent!!.inject(this)

        // android data binding
        _activityListItemBinding = DataBindingUtil.setContentView(this, R.layout.activity_list_item)

        _activityListItemBinding!!.lifecycleOwner = this

        // set viewmodel for databiding
        _activityListItemBinding!!.weatherInCityUIModelForFiveDaysViewModel =
            _weatherInCityUIModelForFiveDaysViewModel

        _weatherInformationUIModelListAdapter =
            WeatherInformationUIModelListAdapter(applicationContext, ArrayList())

        // set adapter for RecyclerView
        _activityListItemBinding!!.rcvItemList.layoutManager =
            LinearLayoutManager(this@WeatherInformationUIModelListActivity)
        _activityListItemBinding!!.rcvItemList.adapter = _weatherInformationUIModelListAdapter

        _weatherInCityUIModelForFiveDaysViewModel.weatherInformationUIModelListOfSelectedDateMutableLiveData.observe(
            this@WeatherInformationUIModelListActivity,
            Observer { weatherInformationUIModels ->
                _weatherInformationUIModelListAdapter!!.setWeatherInformationUIModelList(
                    weatherInformationUIModels
                )
            })

    }

    companion object {

        // intent contain source context and destination context (this Activity) for Navigator
        fun getCallingIntent(_context: Context): Intent {
            return Intent(_context, WeatherInformationUIModelListActivity::class.java)
        }
    }
}
